lib/data/external/AwsClient.js

```diff
@@ -111,6 +113,89 @@ class AwsClient {
             });
         });
     }
+
+    createMPU(key, metaHeaders, websiteRedirectHeader, log, callback) {
+        const bucket = this._awsBucketName;
+        const awsKey = this._createAwsKey(bucket, key, this._bucketMatch);
+        const params = { Bucket: bucket, Key: awsKey,
+            WebsiteRedirectLocation: websiteRedirectHeader,
+            Metadata: metaHeaders };
+        return this._client.createMultipartUpload(params, (err, mpuResObj) => {
+            if (err) {
+                log.error('err from data backend',
+                { error: err, dataStore: this._dataStoreName });
+                return callback(errors.InternalError);
+            }
+            return callback(null, mpuResObj);
+        });
+    }
+
+    uploadPart(stream, hashedStream, size, key, uploadId, partNumber,
+    log, callback) {
+        stream.pipe(hashedStream);
+        const bucket = this._awsBucketName;
+        const awsKey = this._createAwsKey(bucket, key, this._bucketMatch);
+        const params = { Bucket: bucket, Key: awsKey, UploadId: uploadId,
+            Body: hashedStream, ContentLength: size,
+            PartNumber: partNumber };
+        return this._client.uploadPart(params, (err, partResObj) => {
+            if (err) {
+                log.error('err from data backend on uploadPart',
+                { error: err, dataStoreName: this._dataStoreName });
+                return callback(errors.InternalError);
+            }
+            const dataRetrievalInfo = {
+                key,
+                dataStoreName: this._dataStoreName,
+                dataStoreETag: partResObj.ETag,
+            };
+            return callback(null, dataRetrievalInfo);
+        });
+    }
+
+    completeMPU(partList, key, uploadId, log, callback) {
+        const bucket = this._awsBucketName;
+        const awsKey = this._createAwsKey(bucket, key, this._bucketMatch);
+        const partArray = [];
+        return async.each(partList, (partObj, callback) => {
```

Comment:
We may not want to use `async.each` here since the iteratee function is synchronous.